github4s/js/src/main/scala/github4s/HttpRequestBuilderExtensionJS.scala

```diff
@@ -0,0 +1,102 @@
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package github4s
+
+import scala.concurrent.Future
+import fr.hmil.roshttp._
+import fr.hmil.roshttp.body.{BodyPart, BulkBodyPart}
+import java.nio.ByteBuffer
+
+import cats.implicits._
+import fr.hmil.roshttp.response.SimpleHttpResponse
+import fr.hmil.roshttp.util.HeaderMap
+import fr.hmil.roshttp.body.Implicits._
+import fr.hmil.roshttp.exceptions.HttpException
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import github4s.GithubResponses.{GHResponse, GHResult, JsonParsingException, UnexpectedException}
+import github4s.GithubDefaultUrls._
+import github4s.Decoders._
+import io.circe.Decoder
+import io.circe.parser._
+import monix.reactive.Observable
+
+import scala.util.{Failure, Success}
+
+case class CirceJSONBody(value: String) extends BulkBodyPart {
+  override def contentType: String     = s"application/json; charset=utf-8"
+  override def contentData: ByteBuffer = ByteBuffer.wrap(value.getBytes("utf-8"))
+}
+
+trait HttpRequestBuilderExtensionJS {
+
+  import monix.execution.Scheduler.Implicits.global
+
+  def userAgent = {
+    val name    = github4s.BuildInfo.name
+    val version = github4s.BuildInfo.version
+    s"$name/$version"
+  }
+
+  implicit def extensionJS: HttpRequestBuilderExtension[SimpleHttpResponse, Future] =
+    new HttpRequestBuilderExtension[SimpleHttpResponse, Future] {
+      def run[A](rb: HttpRequestBuilder[SimpleHttpResponse, Future])(
+          implicit D: Decoder[A]): Future[GHResponse[A]] = {
+        val request = HttpRequest(rb.url)
+          .withMethod(Method(rb.httpVerb.verb))
+          .withQueryParameters(rb.params.toSeq: _*)
+          .withHeader("content-type", "application/json")
+          .withHeader("user-agent", userAgent)
+          .withHeaders(rb.authHeader.toList: _*)
+          .withHeaders(rb.headers.toList: _*)
+
+        (rb.data match {
+          case Some(d) ⇒
+            request.send(CirceJSONBody(d))
+          case _ ⇒ request.send()
+        }).map(r => toEntity[A](r)).recoverWith[GHResponse[A]] {
+          case e => Future.successful(Either.left(UnexpectedException(e.getMessage)))
+        }
+      }
+    }
+
+  def toEntity[A](response: SimpleHttpResponse)(implicit D: Decoder[A]): GHResponse[A] =
+    response match {
+      case r if r.statusCode < 400 ⇒
```

Comment:
Maybe we could move somewhere here: `github4s.HttpClient.HttpStatus`, together with `github4s.HttpClient.HttpVerb`.

Does it make sense?